oauthproxy.go

```diff
@@ -794,18 +794,19 @@ func (p *OAuthProxy) doOAuthStart(rw http.ResponseWriter, req *http.Request, ove
 	}
 
 	callbackRedirect := p.getOAuthRedirectURI(req)
-	loginURL := p.provider.GetLoginURL(
-		callbackRedirect,
-		encodeState(csrf.HashOAuthState(), appRedirect),
-		csrf.HashOIDCNonce(),
-		extraParams,
-	)
+	hashOAuthState := csrf.HashOAuthState()
+	loginURL := p.provider.GetLoginURL(callbackRedirect, encodeState(hashOAuthState, appRedirect),
+		csrf.HashOIDCNonce(), extraParams)
 
-	if _, err := csrf.SetCookie(rw, req); err != nil {
+	csrfCookie, err := csrf.SetCookie(rw, req)
+	if err != nil {
 		logger.Errorf("Error setting CSRF cookie: %v", err)
 		p.ErrorPage(rw, req, http.StatusInternalServerError, err.Error())
 		return
 	}
+	// There are a lot of issues opened complaining about missing CSRF cookies.
+	// Try to log the INs and OUTs of OAuthProxy, to be easier to analyse these issues.
+	logger.Printf("CSRF cookie %v was sent in request (state=%v).", csrfCookie.Name, hashOAuthState)
```

Comment:
Do we really want to print this every time a login flow is initiated? I think this is where we need to start doing levelled logging

Reply:
I've removed it. 
I was trying to log the INs and OUTs of CSRF header, since there are lot of complains with this issue. I assume that this issue can occur due to bad Oauth2proxy configuration or due to network policies external to Oauth2proxy knowledge.
How can we help users on this topic?